### PR TITLE
Cleanup responses from text

### PR DESCRIPTION
**Pull request by ayush1999**
Added a function to cleanup responses from text comments.

### CODE REVIEW

**Comment by ayush1999** [comment]
Results:
Before changes:
```
ayush99@ayush99:~/GitHub/bugbug$ python run.py --train --goal=bug
(11173, 438922) (11173,)
(1882, 438922) (1882,)
(1118, 438922) (1118,)
CV Accuracy: 0.92 (+/- 0.02)
```
After changes:
```
ayush99@ayush99:~/GitHub/bugbug$ python run.py --train --goal=bug
(11173, 438922) (11173,)
(1882, 438922) (1882,)
(1118, 438922) (1118,)
CV Accuracy: 0.93 (+/- 0.02)
```

**Comment by marco-c** [comment]
The `regression` case is a little better without this, so maybe we have to make it optional.
Let's keep this open for a while until we support making cleanup steps optional.

**Comment by ayush1999** [comment]
There is another similar case for removing code snippets from comments. Should I create a PR for that? (Or we should deal with that later?)

**Comment by ayush1999** [comment]
@marco-c Code snippets, similar to responses, start with `> +` to `\n` . Since this PR removes all text between `>` to `\n`, code snippets are also removed.

**Comment by ayush1999** [comment]
Should I change the function name to something more generic?

**Comment by marco-c** [comment]
> @marco-c Code snippets, similar to responses, start with > + to \n . Since this PR removes all text between > to \n, code snippets are also removed.

Ah no, that's just one of the ways people write code snippets, but it is not a rule.

> Should I change the function name to something more generic?

No, it's OK.

**Comment by ayush1999** [comment]
Ok, I'll let this be open for now.

**Comment by ayush1999** [comment]
@marco-c I've identified a few cases where this works well. Show I add these changes only for those cases? (Similar to #56)

**Comment by marco-c** [comment]
> @marco-c I've identified a few cases where this works well. Show I add these changes only for those cases?

Yes, which cases?

**Comment by ayush1999** [comment]
It works well for bugs and qaneeded. For others, not so much. Should I just add the function for now, similar to the other PRs?

**Comment by marco-c** [comment]
Yes

File: bugbug/bug_features.py [hunk 1]
```diff
@@ -134,6 +134,10 @@ def cleanup_fileref(text):
     return re.sub(r'\w+\.py\b|\w+\.json\b|\w+\.js\b|\w+\.jsm\b|\w+\.html\b|\w+\.css\b|\w+\.c\b|\w+\.cpp\b|\w+\.h\b', '__FILE_REFERENCE__', text)
 
 
+def cleanup_responses(text):
+    return re.sub('>.*?\\n', ' ', text)
```
Comment (marco-c) [inline bugbug/bug_features.py]: Wouldn't `>.*` or `>[^\n]+` be better? After matching `>`, you'd match all characters until a line terminator.

**Comment by ayush1999** [comment]
Using these will not remove the `\n`. Is that what we want? (Won't it be better to remove `\n` too? Otherwise we'd get a series of consecutive `\n` after the cleanup.)

**Comment by marco-c** [comment]
It's fine if we have multiple `\n`, as they are skipped anyway.

Also, if a `>` line doesn't end with `\n` (e.g. if it's the last line), your regex wouldn't match it. Try to add a test containing a `>` line at the end without `\n`.

**Pull request merged**